src/test/shared/defaultAwsContext.test.ts

```diff
@@ -153,6 +163,56 @@ describe('DefaultAwsContext', function () {
         })
     })
 
+    it('sets default region to last region from prompter', async function () {
+        const regions = [
+            { id: 'us-west-2', name: 'PDX' },
+            { id: 'us-east-1', name: 'IAD' },
+            { id: 'foo-bar-1', name: 'FOO' },
+        ]
+        const p = createRegionPrompter(regions, {
+            title: 'Select regionnnn',
+            buttons: createCommonButtons('https://aws.amazon.com/'),
+            defaultRegion: 'foo-bar-1',
+        })
+        const tester = createQuickPickTester(p)
+        const selection = regions[0]
+        tester.assertItems(['FOO', 'PDX', 'IAD'])
+        tester.acceptItem({
+            label: selection.name,
+            detail: selection.id,
+            data: selection,
+            skipEstimate: true,
+            description: '',
+        })
+        await tester.result(selection)
+
+        const testContext = await FakeExtensionContext.create()
+        const awsContext: AwsContext = new DefaultAwsContext(testContext)
+        const result = awsContext.guessDefaultRegion()
```

Comment:
Since we reset state after each test I know that the original region, before the region prompter, should be `us-east-1` and the value after region prompter should be 'us-west-2' since thats the selection. Thus, I should be able to check this by asserting they are never equal.

Reply:
It seems like I can test this by making sure the value changes after the region prompter. To be safe, I think I should also use a fake region as the selection, that way we know the region will change and not somehow run into the scenario that the default and the selection were the same.